src/Nethermind/Nethermind.Merge.Plugin/GC/GCKeeper.cs

```diff
@@ -0,0 +1,180 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Diagnostics;
+using System.Runtime;
+using System.Threading;
+using System.Threading.Tasks;
+using FastEnumUtility;
+using Nethermind.Core.Extensions;
+using Nethermind.Logging;
+
+namespace Nethermind.Merge.Plugin.GC;
+
+public class GCKeeper
+{
+    private static ulong _forcedGcCount = 0;
+    private readonly IGCStrategy _gcStrategy;
+    private readonly ILogger _logger;
+    private static readonly long _defaultSize = 512.MB();
+    private Task _gcScheduleTask = Task.CompletedTask;
+
+    public GCKeeper(IGCStrategy gcStrategy, ILogManager logManager)
+    {
+        _gcStrategy = gcStrategy;
+        _logger = logManager.GetClassLogger<GCKeeper>();
+    }
+
+    public IDisposable TryStartNoGCRegion(long? size = null)
+    {
+        size ??= _defaultSize;
+        var priorLatencyMode = System.Runtime.GCSettings.LatencyMode;
+        //if (_gcStrategy.CanStartNoGCRegion())
+        if (priorLatencyMode != GCLatencyMode.SustainedLowLatency)
```

Comment:
We could have some strategies here:
1. NoGCRegionStrategy
2. SustainedLowLatencyStrategy

and invoke one or the other - or both!

Reply:
Sounds like follow up post bug?